shared-bindings/mipidsi/Bus.c

```diff
@@ -0,0 +1,96 @@
+// This file is part of the CircuitPython project: https://circuitpython.org
+//
+// SPDX-FileCopyrightText: Copyright (c) 2025 Scott Shawcroft for Adafruit Industries
+//
+// SPDX-License-Identifier: MIT
+
+#include "py/obj.h"
+#include "py/objproperty.h"
+#include "py/runtime.h"
+
+#include "shared-bindings/mipidsi/Bus.h"
+#include "shared-bindings/util.h"
+#include "shared/runtime/context_manager_helpers.h"
+
+//| class Bus:
+//|     def __init__(
+//|         self,
+//|         *,
+//|         frequency: int = 500_000_000,
+//|         num_lanes: int = 2,
+//|     ) -> None:
+//|         """Create a MIPI DSI Bus object.
+//|
+//|         This creates a DSI bus interface. The specific pins used are determined by the board.
+//|         DSI supports 1-4 data lanes.
+//|
+//|         :param int frequency: the high speed clock frequency in Hz (default 500 MHz)
+//|         :param int num_lanes: the number of data lanes to use (default 2, range 1-4)
+//|         """
+//|
+
+static mp_obj_t mipidsi_bus_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
+    enum { ARG_frequency, ARG_num_lanes };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_frequency, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 500000000} },
+        { MP_QSTR_num_lanes, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 2} },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all_kw_array(n_args, n_kw, all_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    mipidsi_bus_obj_t *self = mp_obj_malloc(mipidsi_bus_obj_t, &mipidsi_bus_type);
```

Comment:
I think this could be with `mp_obj_malloc_with_finaliser` if we add `__del__` below.

Reply:
I statically allocated it instead because we'll only ever have one and multiple displays may use it. One may live outside the VM as a "display bus" from a framebuffer display.